seurat/CellClassificationFXs.R

```diff
@@ -0,0 +1,605 @@
+    }
+
+    gc(verbose = FALSE)
+    return(object)
+
+  } else {
+    return(genes.scores.use)
+  }
+
+
+
+}
+
+
+ClassifyCellsCustom_SERIII <- function(Classifier.rds.path = "",
+                                       ClassifierNames="",
+                                       testing.data, log10T=T, returnTraining=F){
+
+  if(file.exists(Classifier.rds.path)){
+
+    MultiClassifierResults <- readRDS(Classifier.rds.path)
+    Available.Classifiers  <- names(MultiClassifierResults)[grepl("classifier", names(MultiClassifierResults))]
+
+    print(Available.Classifiers)
+
+    if(ClassifierNames[1]=="") ClassifierNames <- Available.Classifiers
+
+    if(length(which(ClassifierNames %in% Available.Classifiers)) < length(ClassifierNames)) {
+      print("Names in ClassifierNames did not match whats available")
+      ClassifierNames <- Available.Classifiers
+    }
+    MultiClassifier        <- MultiClassifierResults[Available.Classifiers]
+
+
+    if(class(testing.data)[1]=="seurat") {
+      print("Converting Seurat Sparse Matrix to one with 0s .... ")
+
+      testing.data <- Matrix::as.matrix(Matrix::t(testing.data@assays$RNA@data))
+
+    } else {
+      if(!(class(testing.data)[1] %in% c("dgCMatrix", "matrix", "Matrix")) ) {
+        print("DGE mat not Seurat or dgCMatrix")
+        print("Make sure columns are cells and rows are genes and convert to expected format for speed")
+        testing.data <- Matrix::as.matrix((testing.data))
+
+      }
+    }
+
+
+    if(log10T) {
+      print("log-transforming")
+      testing.data <- log10(testing.data+1)
+    }
+
+
+    print("Starting Classification")
+
+    testing.data.yhat <- data.frame(lapply(Available.Classifiers, function(ClassifX){
+
+      print(ClassifX)
+
+      tempClassifier <- MultiClassifier[[ClassifX]]
+
+      tempClassifier$coefnames[!(tempClassifier$coefnames %in% colnames(testing.data))]
+
+
+      y.hat <- predict(tempClassifier, newdata = testing.data[,])
+
+
+
+
+      return(y.hat)
+    }))
+
+    colnames(testing.data.yhat) <- Available.Classifiers
+
+    NotLevelName <- levels(testing.data.yhat[,1])[grep("Not", levels(testing.data.yhat[,1]))]
+
+    testing.data.yhat$CountNot <- apply(testing.data.yhat, 1, function(x) sum(grepl(NotLevelName, x)))
+    testing.data.yhat$CountTot <- rep(length(Available.Classifiers), nrow(testing.data.yhat))
+
+    testing.data.yhat$NotProb <- testing.data.yhat$CountNot/testing.data.yhat$CountTot
+
+
+
+    rownames(testing.data.yhat) <- rownames(testing.data)
+
+
+    return(list(yhat.DF = testing.data.yhat,
+                classification.levels = levels(testing.data.yhat[,1]),
+                Available.Classifiers = Available.Classifiers,
+                log10T = log10T,
+                DGEcolNames = colnames(testing.data),
+                DGErowNames = rownames(testing.data),
+                MultiClassifierResults = ifelse(returnTraining, MultiClassifierResults, "NULL")))
+
+  } else {
```

Comment:
Why not place this check first, and abort if it fails?  this saves having to wrap the entire first part of the method in an if()?